system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp

```diff
@@ -170,35 +180,69 @@ TEST_F(PeriodicMeasurementTestFixure, TestStartAndStop) {
     test_constants::kTestDuration.count() / kDontPublishDuringTest.count(), times_published);
 }
 
-TEST_F(PeriodicMeasurementTestFixure, TestConstructorInputValidation) {
-  ASSERT_THROW(TestPeriodicMeasurementNode("throw",
-    std::chrono::milliseconds{-1},
-    "",
-    kDontPublishDuringTest), std::invalid_argument);
-
-  // bad measurement period
-  ASSERT_THROW(TestPeriodicMeasurementNode("throw",
-    std::chrono::milliseconds{-1},
-    "",
-    test_constants::kPublishPeriod), std::invalid_argument);
-
-  // bad node name
-  ASSERT_THROW(TestPeriodicMeasurementNode("throw",
-    test_constants::kMeasurePeriod,
-    kTestTopic,
-    std::chrono::milliseconds{-1}), std::invalid_argument);
-
-  // invalid publish period
-  ASSERT_THROW(TestPeriodicMeasurementNode("throw",
-    std::chrono::milliseconds{2},
-    kTestTopic,
-    std::chrono::milliseconds{1}), std::invalid_argument);
-
-  // invalid node name
-  ASSERT_THROW(TestPeriodicMeasurementNode("",
-    std::chrono::milliseconds{2},
-    kTestTopic,
-    std::chrono::milliseconds{1}), std::invalid_argument);
+TEST_F(PeriodicMeasurementTestFixure, TestConstructorMeasurementPeriodValidation1) {
+  rclcpp::NodeOptions options;
+  options.append_parameter_override(
+    system_metrics_collector::collector_node_constants::kCollectPeriodParam,
+    std::chrono::milliseconds{-1}.count());
+  options.append_parameter_override(
+    system_metrics_collector::collector_node_constants::kPublishPeriodParam,
+    kDontPublishDuringTest.count());
+
+  ASSERT_THROW(TestPeriodicMeasurementNode("throw", options),
+    rclcpp::exceptions::InvalidParameterValueException);
+}
+
+TEST_F(PeriodicMeasurementTestFixure, TestConstructorMeasurementPeriodValidation2) {
```

Comment:
How is this not identical to TestConstructorMeasurementPeriodValidation1?

Reply:
I have removed `TestConstructorMeasurementPeriodValidation1`.